### PR TITLE
Control examples

### PR DESCRIPTION
**Pull request by Checkmate50**
Removed the `node(...)` and `none(...)` syntax from the assembly (finally).  Includes a few related bugfixes for the lowering pass.

Also includes a few more examples of interesting control flow, notably `basics/alloc_reuse` and `control_flow/call_sequence`.  Do note that `control_flow/max_of_min` is not typechecking, so I'll fix that on the next bit of work, I just wanted to get existing changes in.

Also note that this PR _does_ break high-level caiman some amount of the time, specifically in two ways:
* Lowering doesn't seem to perfectly account for the input/output changes above, which should be a simple fix
* Lowering previously relied on the input/output tags to encode information, which now requires an explicit variable.  This means that tests which rely on input/node or node/output interactions fail to typecheck

### CODE REVIEW

File: high-level-caiman/src/lower/lower_schedule.rs [hunk 1]
```diff
@@ -521,17 +504,26 @@ fn lower_block(funclet: &Funclet<'_>) -> asm::Funclet {
...
             args: funclet.inputs(),
             ret: funclet.outputs(),
             binding: asm::FuncletBinding::ScheduleBinding(asm::ScheduleBinding {
-                implicit_tags: None,
-                value: Some(funclet.specs().value.clone()),
-                timeline: Some(funclet.specs().timeline.clone()),
-                spatial: Some(funclet.specs().spatial.clone()),
+                implicit_tags: (implicit_default.clone(), implicit_default),
+                meta_map: asm::MetaMapping {
+                    value: (MetaId(META_VALUE.to_string()), funclet.specs().value.clone()),
```
Comment (stephenverderame) [inline high-level-caiman/src/lower/lower_schedule.rs:523]: Should these all be mapping to the value spec?

File: high-level-caiman/src/lower/lower_schedule.rs [hunk 2]
```diff
@@ -433,30 +420,30 @@ fn lower_select(guard_name: &str, tags: &TripleTag, temp_id: usize, f: &Funclet<
             },
         })),
         Some(asm::Command::TailEdge(asm::TailEdge::ScheduleSelect {
-            value_operation: Some(
+            operations: Some(vec![
```
Comment (stephenverderame) [inline high-level-caiman/src/lower/lower_schedule.rs:423]: Just curious, but does order matter, or is the assembly using the meta-variables to determine which tag is for which spec.

**Review by stephenverderame — APPROVED** [review]
I think the frontend changes seem reasonable enough to merge right now. I did leave two small comments. My only thought is that the frontend tests do test more behavior than the current assembly tests, but if you're comfortable, that's fine with me

File: high-level-caiman/src/lower/lower_schedule.rs (hunk 1 above)
Comment (Checkmate50) [reply high-level-caiman/src/lower/lower_schedule.rs:523]: oops, good catch!  That's probably causing one of the bugs that showed up, thanks!

File: high-level-caiman/src/lower/lower_schedule.rs (hunk 2 above)
Comment (Checkmate50) [reply high-level-caiman/src/lower/lower_schedule.rs:423]: The assembly uses the meta-variables to determine which tag is for which spec, yep!

**Pull request merged**